projects/tcpdump/pretty_print_packet.c

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 ISP RAS
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//			http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#include <pcap/pcap.h>
+#include <netdissect-stdinc.h>
+#include <netdissect.h>
+#include <print.h>
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+static uint64_t packets_captured = 0;
+
+void print_packet(unsigned char *user, const struct pcap_pkthdr *h, const unsigned char *sp) {
+	++packets_captured;
+	pretty_print_packet((netdissect_options *)user, h, sp, packets_captured);
+}
+
+FILE *bufferToFile(const uint8_t *Data, size_t Size) {
+	FILE *fd;
+	fd = tmpfile();
```

Comment:
shouldn't we remove temporary file later?

Reply:
I found this in `man tmpfile`: The tmpfile() function opens a unique temporary file in binary read/write (w+b) mode.  The file will be automatically deleted when it is closed or the program terminates.